### ISSUE TITLE
Problem with `dr.backward`, when using `@dr.wrap_ad`: 

### ISSUE THREAD

**Issue by maxfrei750** [mitsuba-renderer/drjit#76]
When porting the [Gradient based optimization tutorial ](https://github.com/mitsuba-renderer/mitsuba-tutorials/blob/b503c58ed863fafb4d4a280cd105dbfaefbcc0c4/inverse_rendering/gradient_based_opt.ipynb) to PyTorch using `@dr.wrap_ad`:

```
import drjit as dr
import mitsuba as mi
import numpy as np
import torch


@dr.wrap_ad(source="drjit", target="torch")
def criterion(image, image_ref):
    return torch.mean(torch.square(image - image_ref))


def test_llvm_ad_rgb_torch():
    rendering_seed = 0
    comparison_spp = 512
    image_size = 128

    mi.set_variant("llvm_ad_rgb")

    scene_dictionary = mi.cornell_box()
    scene_dictionary["sensor"]["film"]["width"] = image_size
    scene_dictionary["sensor"]["film"]["height"] = image_size
    scene_dictionary["integrator"] = dict(type="prb")
    test_scene = mi.load_dict(scene_dictionary)
    image_reference = mi.render(test_scene, seed=rendering_seed, spp=comparison_spp)

    parameters = mi.traverse(test_scene)

    key = "red.reflectance.value"

    color_reference = parameters[key]

    # Set another color value and update the scene
    parameters[key] = mi.Color3f(0.01, 0.2, 0.9)
    parameters.update()

    image_init = mi.render(test_scene, seed=rendering_seed, spp=128)
    mi.util.convert_to_bitmap(image_init)

    optimizer = mi.ad.Adam(lr=0.05)
    optimizer[key] = parameters[key]
    parameters.update(optimizer)

    for _ in range(50):
        # Perform a (noisy) differentiable rendering of the scene
        image = mi.render(test_scene, parameters, spp=4)

        # Evaluate the objective function from the current rendered image
        loss = criterion(image, image_reference)

        # Backpropagate through the rendering process
        dr.backward(loss)

        # Optimizer: take a gradient descent step
        optimizer.step()

        # Post-process the optimized parameters to ensure legal color values.
        optimizer[key] = dr.clamp(optimizer[key], 0.0, 1.0)

        # Update the scene state to the new optimized values
        parameters.update(optimizer)

    image_final = mi.render(test_scene, seed=rendering_seed, spp=comparison_spp)

    color_restored = parameters[key]

    np.testing.assert_allclose(color_reference, color_restored, atol=0.01)
    np.testing.assert_allclose(image_final, image_reference, atol=0.01)
    print("Success!")


if __name__ == "__main__":
    test_llvm_ad_rgb_torch()
```

I receive the following error:
```
Exception has occurred: TypeError
__init__(): incompatible constructor arguments. The following argument types are supported:
    1. drjit.llvm.TensorXf()
    2. drjit.llvm.TensorXf(array: object)
    3. drjit.llvm.TensorXf(array: drjit.llvm.Float)
    4. drjit.llvm.TensorXf(array: drjit.llvm.Float, shape: List[int])
    5. drjit.llvm.TensorXf(arg0: drjit.llvm.TensorXi)
    6. drjit.llvm.TensorXf(arg0: drjit.llvm.TensorXu)
    7. drjit.llvm.TensorXf(arg0: drjit.llvm.TensorXi64)
    8. drjit.llvm.TensorXf(arg0: drjit.llvm.TensorXu64)
    9. drjit.llvm.TensorXf(arg0: drjit.llvm.TensorXf)
    10. drjit.llvm.TensorXf(arg0: drjit.llvm.TensorXf64)

Invoked with: None, (128, 128, 3)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4055, in accum_grad
    src = t(src, dst.shape)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4066, in accum_grad
    accum_grad(dst[i], src[i] if vs else src)
  File "/src/mitsuba3/build/python/drjit/router.py", line 5490, in set_grad_in
    _dr.accum_grad(self.inputs[name], value)
  File "/src/mitsuba3/build/python/drjit/router.py", line 5868, in backward
    self.set_grad_in('args', args_grad)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4341, in traverse
    dtype.traverse_(mode, flags)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4460, in backward_from
    traverse(ta, _dr.ADMode.Backward, flags)
  File "/src/mitsuba3/build/python/drjit/router.py", line 4514, in backward
    backward_from(arg, flags)
  File "/workspace/development/test_llvm_ad_rgb_torch.py", line 51, in test_llvm_ad_rgb_torch
    dr.backward(loss)
  File "/workspace/development/test_llvm_ad_rgb_torch.py", line 72, in <module>
    test_llvm_ad_rgb_torch()
```

This is strange, since it used to work with previous versions of the `wrap_ad` branch. Has something changed about the usage?

**Comment by Speierers** [comment]
Thanks @maxfrei750 for reporting this. It will be fixed in the next release: [805f4ce](https://github.com/mitsuba-renderer/drjit/commit/805f4cef644e55935e8e55228e2b8f224fdc054b)
